recipes/libtar/all/conanfile.py

```diff
@@ -24,76 +29,84 @@ class LibTarConan(ConanFile):
         "with_zlib": True,
     }
 
-    _autotools = None
-
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
-
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
 
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
-        del self.settings.compiler.libcxx
-        del self.settings.compiler.cppstd
+            try:
+                del self.options.fPIC
+            except Exception:
+                pass
+        try:
+            del self.settings.compiler.libcxx
+        except Exception:
+            pass
+        try:
+            del self.settings.compiler.cppstd
+        except Exception:
+            pass
+
+    def layout(self):
+        basic_layout(self, src_folder="src")
 
     def requirements(self):
         if self.options.with_zlib:
-            self.requires("zlib/1.2.11")
+            self.requires("zlib/1.2.12")
 
     def validate(self):
-        if self.settings.os == "Windows":
+        if self.info.settings.os == "Windows":
```

Comment:
I don't see this documented until version 1.51.0 of Conan [here](https://docs.conan.io/en/1.51/reference/conanfile/methods.html#validate-build).

Reply:
It's documented since 1.50.0: https://docs.conan.io/en/1.50/migrating_to_2.0/recipes.html#the-validate-method But it's much older as `self.info` is available for package_id.